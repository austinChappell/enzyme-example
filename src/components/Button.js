import React from 'react';
import styled from 'styled-components';

import { COLORS } from '../constants/styles';

const StyledButton = styled.button({
  background: `linear-gradient(120deg, ${COLORS.primaryMain}, ${COLORS.secondaryMain})`,
  borderRadius: 8,
  color: '#fff',
  cursor: 'pointer',
  fontSize: 20,
  padding: 8,
  '&:hover': {
    background: `linear-gradient(120deg, ${COLORS.primaryDark}, ${COLORS.secondaryDark})`,
  }
});

const Button = ({
  children,
  onClick,
}) => (
  <StyledButton onClick={onClick}>
    {children}
  </StyledButton>
)

export default Button;
