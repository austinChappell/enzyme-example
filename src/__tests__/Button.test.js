import React from 'react';
import { mount, render, shallow } from 'enzyme'

import Button from '../components/button';

describe('The Button Component', () => {
  it('should match snapshot', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(
      <Button onClick={mockClick}>
        Click Me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('should trigger the onClick prop', () => {
    const mockClick = jest.fn();
    const wrapper = mount(
      <Button onClick={mockClick}>
        Click Me
      </Button>
    );
    wrapper.find('button').simulate('click')
    expect(mockClick).toHaveBeenCalled();
  })

  it('should change color on hover', () => {
    const mockClick = jest.fn();
    const wrapper = render(
      <Button onClick={mockClick}>
        Click Me
      </Button>
    );
    console.log('background', wrapper.get(0))
  })
});

