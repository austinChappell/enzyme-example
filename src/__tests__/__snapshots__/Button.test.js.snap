// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button Component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onClick={[MockFunction]}
  >
    Click Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Click Me",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": "Click Me",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "background: linear-gradient(120deg, #5C61DE, #DE6D25); border-radius: 8px; color: #fff; cursor: pointer; font-size: 20px; padding: 8px; &:hover {
  background: linear-gradient(120deg, #42459E, #8C4517);
}",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Click Me",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": "Click Me",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "background: linear-gradient(120deg, #5C61DE, #DE6D25); border-radius: 8px; color: #fff; cursor: pointer; font-size: 20px; padding: 8px; &:hover {
  background: linear-gradient(120deg, #42459E, #8C4517);
}",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
